docs/PREFERENCES.md

```diff
@@ -21,22 +21,23 @@ Preferences can be controlled and modified in the settings window or via the `pr
+| textDirection                      | String  | ltr                | The writing text direction, optional value: `ltr` or `rtl`                                                                                                                     |
+| codeFontSize                       | Number  | 14                 | Font size on code block, the range is 12 ~ 28                                                                                                                                  |
+| codeFontFamily                     | String  | `DejaVu Sans Mono` | Code font family                                                                                                                                                               |
+| trimUnnecessaryCodeBlockEmptyLines | Boolean | true               | Whether to trim the beginning and end empty line in Code block                                                                                                                 |
+| hideQuickInsertHint                | Boolean | false              | Hide hint for quickly creating paragraphs                                                                                                                                      |
 | imageDropAction                    | String  | folder             | The default behavior after paste or drag the image to Mark Text, upload it to the image cloud (if configured), move to the specified folder, insert the path          |
-| defaultEncoding                    | String  | `utf8`             | The default file encoding                                                                                                                                             |
-| autoGuessEncoding                  | Boolean | true               | Try to automatically guess the file encoding when opening files                                                                                                       |
-| hideLinkPopup                      | Boolean | false              | It will not show the link popup when hover over the link if set `hideLinkPopup` to true                                                                               |
+| defaultEncoding                    | String  | `utf8`             | The default file encoding                                                                                                                                                      |
+| autoGuessEncoding                  | Boolean | true               | Try to automatically guess the file encoding when opening files                                                                                                                |
+| trimTrailingNewline                | Enum    | `2`                | Ensure a single trailing newline or whether trailing newlines should be removed: `0`: ensure single newline, `1`: trim all trailing newlines, `2`: auto detect, `3`: disabled. |
```

Comment:
What's the meaning of `auto detect` here?

Reply:
That we detect whether a single new line should be ensured or no newlines should be added with fallback to the current behavior.